src/ShieldAPI.php

```diff
@@ -0,0 +1,634 @@
+            endpoint: $endpoint,
+            parameters: [$id],
+            body: BodyContentHelper::getBody($body),
+        );
+    }
+
+    /**
+     * @throws ClientExceptionInterface
+     * @throws Exception\BunnyClientResponseException
+     * @throws Exception\JSONException
+     * @return BunnyClientResponseInterface
+     * @param int $id
+     */
+    public function deleteCustomWafRule(int $id): BunnyClientResponseInterface
+    {
+        $endpoint = new DeleteCustomWAFRule();
+
+        return $this->client->request(
+            endpoint: $endpoint,
+            parameters: [$id],
+        );
+    }
+
+    /**
+     * @throws ClientExceptionInterface
+     * @throws Exception\BunnyClientResponseException
+     * @throws Exception\JSONException
+     * @throws Exception\InvalidTypeForKeyValueException
+     * @throws Exception\InvalidTypeForListValueException
+     * @throws Exception\ParameterIsRequiredException
+     * @return BunnyClientResponseInterface
+     * @param array<string,mixed> $body
+     */
+    public function createCustomWafRule(
+        array $body,
+    ): BunnyClientResponseInterface {
+        $endpoint = new CreateCustomWAFRule();
+
+        ParameterValidator::validate($body, $endpoint->getBody());
+
+        return $this->client->request(
+            endpoint: $endpoint,
+            body: BodyContentHelper::getBody($body),
+        );
+    }
+
+    /**
+     * @throws ClientExceptionInterface
+     * @throws Exception\BunnyClientResponseException
+     * @throws Exception\JSONException
+     * @return BunnyClientResponseInterface
+     */
+    public function listWafProfiles(): BunnyClientResponseInterface
+    {
+        $endpoint = new ListWafProfiles();
+
+        return $this->client->request(
+            endpoint: $endpoint,
+        );
+    }
+
+    /**
+     * @throws ClientExceptionInterface
+     * @throws Exception\BunnyClientResponseException
+     * @throws Exception\JSONException
+     * @return BunnyClientResponseInterface
+     */
+    public function listWafEnums(): BunnyClientResponseInterface
+    {
+        $endpoint = new ListWAFEnums();
+
+        return $this->client->request(
+            endpoint: $endpoint,
+        );
+    }
+
+    /**
+     * @throws ClientExceptionInterface
+     * @throws Exception\BunnyClientResponseException
+     * @throws Exception\JSONException
+     * @return BunnyClientResponseInterface
+     */
+    public function listWafEngineConfiguration(): BunnyClientResponseInterface
+    {
+        $endpoint = new ListWAFEngineConfiguration();
+
+        return $this->client->request(
+            endpoint: $endpoint,
+        );
+    }
+
+    /**
+     * @throws ClientExceptionInterface
+     * @throws Exception\BunnyClientResponseException
+     * @throws Exception\JSONException
+     * @return BunnyClientResponseInterface
+     */
+    public function listDdosEnums(): BunnyClientResponseInterface
+    {
+        $endpoint = new ListDDOsEnums();
```

Comment:
Not sure why phpstan did not recognize this as incorrect casing.

```suggestion
        $endpoint = new ListDDoSEnums();
```